artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -3716,6 +3730,16 @@ private boolean deliver(final MessageReference ref) {
       }
    }
 
+   private void accountForChangeInMemoryEstimate(final MessageReference ref, final int existingMemoryEstimate) {
+      final int delta = ref.getMessageMemoryEstimate() - existingMemoryEstimate;
+      if (delta > 0) {
+         PagingStore pageStore = ref.getOwner();
+         if (pageStore != null) {
+            pageStore.addSize(delta);
```

Comment:
2 observations:

1. what happen to core messages here?
2. this method could become static

Reply:
thanks franz,
a core message won't have any delta that I am aware of, it's memory estimate does not change, but you were in that space recently, does it change?

Reply:
yes to the static, let me fix that, thanks!